src/components/form/GenericField/GenericField.tsx

```diff
@@ -0,0 +1,388 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import { Input } from 'semantic-ui-react';
+import Dropdown from 'components/elements/Dropdown';
+import ErrorMessage from 'components/elements/ErrorMessage';
+import Checkbox from 'components/form/Checkbox';
+import FormField from '../Form/FormField';
+
+import GenericFieldEnum from './GenericFieldEnum';
+
+type GenericFieldItemType = {
+    value: boolean | number | string;
+    text?: string | number;
+    name?: string | number;
+};
+
+interface GenericFieldProps {
+    /**
+     * field's label to show above the field
+     */
+    label: string;
+
+    /**
+     * name of the input field
+     */
+    name: string;
+
+    /**
+     * React element type (ie. MyComponent) to be used as input field,
+     * it must provide at least the following props: 'name', 'value', 'onChange',
+     * custom props, will be passed through
+     * (valid only for CUSTOM_TYPE)
+     */
+    component?: any;
```

Comment:
I think it can only be React element, so maybe something like:
```suggestion
    component?: ReactElement<{name: string; value: any; onChange: typeof handleInputChange}, any>;
```

I didn't test it, it's just a general concept.